tests/unit/parser.js

```diff
@@ -6001,6 +6001,20 @@ exports["test for line breaks with 'yield'"] = function (test) {
     .addError(13, "Missing semicolon.");
 
   run.test(code, {moz: true});
+
+  var code2 = [
+    "function* gen() {",
+    "  yield",
+    "  fn();",
```

Comment:
Could you add another statement for `yield*`? Something like

```diff
--- a/tests/unit/parser.js
+++ b/tests/unit/parser.js
@@ -6006,13 +6006,18 @@ exports["test for line breaks with 'yield'"] = function (test) {
     "function* gen() {",
     "  yield",
     "  fn();",
+    "  yield*",
+    "  fn();",
     "}"
   ];
 
-  TestRun(test, "gh-2530").test(code2, { esnext: true, undef: false, asi: true });
+  TestRun(test, "gh-2530")
+    .addError(5, "Bad line breaking before 'fn'.")
+    .test(code2, { esnext: true, undef: false, asi: true });
 
   TestRun(test, "gh-2530")
     .addError(3, "Bad line breaking before 'fn'.")
+    .addError(5, "Bad line breaking before 'fn'.")
     .test(code2, { esnext: true, undef: false });
 ```

It's not a problem with the patch, but it's a likely source of regression, so I'd like an explicit test for it.